### PR TITLE
SDI-355 add current flag to get incentive and add new get current inc…

### PR DESCRIPTION
**Pull request by lbennett-moj**
…entive endpoint

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/IncentivesService.kt [hunk 1]
```diff
@@ -35,19 +35,35 @@ class IncentivesService(
         offenderBooking = offenderBooking,
         sequence = incentiveSequence
       )
-    )?.let { mapIncentiveModel(it) }
+    )?.let {
+      // determine if this incentive is the current IEP for the booking
+      val currentIep =
+        incentiveRepository.findFirstById_offenderBookingOrderByIepDateDescId_SequenceDesc(offenderBooking)
+          ?.let { current -> current == it } ?: false
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/IncentivesService.kt:42]: i *think possibly this is equivalent: 
val currentIep = it ==
        incentiveRepository.findFirstById_offenderBookingOrderByIepDateDescId_SequenceDesc(offenderBooking)

**Review by steverendell — DISMISSED** [review]
lgtm with a comment

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/IncentivesService.kt (hunk 1 above)
Comment (lbennett-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/IncentivesService.kt:42]: yes - thanks, makes it simpler

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**